dashboard/app/models/user.rb

```diff
@@ -237,6 +241,20 @@ def admins_must_be_teachers_without_followeds
     end
   end
 
+  def email
+    unless provider && provider == 'migrated'
+      return self[:email]
+    end
+    primary_email.nil? ? '' : primary_email.email
```

Comment:
`primary_email.try(:email)`

Reply:
or the new ruby 5 safe navigation operator!
`primary_email&.email`